File: backend/app/schemas/recovery.py
from typing import List

from pydantic import BaseModel, Field


class RecoveryLogRequest(BaseModel):
    """Payload for logging a daily recovery check-in.

    Quality/stress/soreness use a 1-5 scale; sleep_hours is in hours.
    """

    sleep_hours: float = Field(..., ge=0, le=24)
    sleep_quality: int = Field(..., ge=1, le=5)
    stress_level: int = Field(..., ge=1, le=5)
    muscle_soreness: int = Field(..., ge=1, le=5)


class RecoveryLogResponse(BaseModel):
    """Response returned after logging recovery."""

    recovery_score: int
    recommendation: str


class RecoveryScoreResponse(BaseModel):
    """Response for today's recovery score."""

    score: int
    recommendation: str
    explanation: str


class RecoveryHistoryItem(BaseModel):
    """A single day's score in the recovery history."""

    date: str
    score: int


class RecoveryHistoryResponse(BaseModel):
    """Response for the recovery history endpoint."""

    history: List[RecoveryHistoryItem]

File: frontend/src/types/recovery.ts
/** Recovery types, aligned with the backend recovery schemas. */

/** Body POSTed to `/recovery/log`. Quality/stress/soreness are 1–5 scales. */
export type RecoveryLogPayload = {
  sleep_hours: number;
  sleep_quality: number;
  stress_level: number;
  muscle_soreness: number;
};

/** `POST /recovery/log` -> data. */
export type RecoveryLogResult = {
  recovery_score: number;
  recommendation: string;
};

/** `GET /recovery/score/today` -> data. */
export type RecoveryScoreData = {
  score: number;
  recommendation: string;
  explanation: string;
};

/** One point in `GET /recovery/history` -> data.history. */
export type RecoveryHistoryItem = {
  date: string;
  score: number;
};
